File: Web/src/api/stock/stoStockLog.ts
import {useBaseApi} from '/@/api/base';

// 库存变动日志接口服务
export const useStoStockLogApi = () => {
	const baseApi = useBaseApi("stoStockLog");
	return {
		// 分页查询库存变动日志
		page: baseApi.page,
		// 查看库存变动日志详细
		detail: baseApi.detail,
		// 新增库存变动日志
		add: baseApi.add,
		// 更新库存变动日志
		update: baseApi.update,
		// 删除库存变动日志
		delete: baseApi.delete,
		// 批量删除库存变动日志
		batchDelete: baseApi.batchDelete,
		// 导出库存变动日志数据
		exportData: baseApi.exportData,
		// 导入库存变动日志数据
		importData: baseApi.importData,
		// 下载库存变动日志数据导入模板
		downloadTemplate: baseApi.downloadTemplate,
	}
}

// 库存变动日志实体
export interface StoStockLog {
	// 主键Id
	id: number;
	// 业务类型(StockIn入库/StockOut出库/Transfer调拨/Adjust调整/Count盘点)
	businessType?: string;
	// 业务单据号
	businessNo?: string;
	// 仓库ID
	warehouseId: number;
	// 仓库编码
	warehouseCode: string;
	// 物料ID
	materialId: number;
	// 物料编码
	materialCode: string;
	// 物料名称
	materialName: string;
	// 变动类型(Increase增加/Decrease减少)
	changeType?: string;
	// 变动数量
	changeQuantity?: number;
	// 变动前数量
	beforeQuantity?: number;
	// 变动后数量
	afterQuantity?: number;
	// 成本单价
	costPrice?: number;
	// 变动金额
	changeAmount?: number;
	// 库位编码
	locationCode: string;
	// 批号
	batchNo: string;
	// 备注
	remark: string;
	// 创建时间
	createTime: string;
	// 更新时间
	updateTime: string;
	// 创建者Id
	createUserId: number;
	// 创建者姓名
	createUserName: string;
	// 修改者Id
	updateUserId: number;
	// 修改者姓名
	updateUserName: string;
}

File: CzhERP.Application/Entity/StoStockLog.cs
namespace CzhERP.Application.Entity;

[Tenant("1300000000001")]
[SugarTable("Sto_StockLog", "库存变动日志")]
public partial class StoStockLog : EntityBase
{
    [Required]
    [SugarColumn(ColumnName = "BusinessType", ColumnDescription = "业务类型(StockIn入库/StockOut出库/Transfer调拨/Adjust调整/Count盘点)", Length = 20)]
    public virtual string BusinessType { get; set; }

    [Required]
    [SugarColumn(ColumnName = "BusinessNo", ColumnDescription = "业务单据号", Length = 50)]
    public virtual string BusinessNo { get; set; }

    [SugarColumn(ColumnName = "WarehouseId", ColumnDescription = "仓库ID")]
    public virtual long? WarehouseId { get; set; }

    [SugarColumn(ColumnName = "WarehouseCode", ColumnDescription = "仓库编码", Length = 50)]
    public virtual string? WarehouseCode { get; set; }

    [SugarColumn(ColumnName = "MaterialId", ColumnDescription = "物料ID")]
    public virtual long? MaterialId { get; set; }

    [SugarColumn(ColumnName = "MaterialCode", ColumnDescription = "物料编码", Length = 50)]
    public virtual string? MaterialCode { get; set; }

    [SugarColumn(ColumnName = "MaterialName", ColumnDescription = "物料名称", Length = 100)]
    public virtual string? MaterialName { get; set; }

    [Required]
    [SugarColumn(ColumnName = "ChangeType", ColumnDescription = "变动类型(Increase增加/Decrease减少)", Length = 20)]
    public virtual string ChangeType { get; set; }

    [Required]
    [SugarColumn(ColumnName = "ChangeQuantity", ColumnDescription = "变动数量")]
    public virtual decimal ChangeQuantity { get; set; }

    [SugarColumn(ColumnName = "BeforeQuantity", ColumnDescription = "变动前数量", DefaultValue = "0")]
    public virtual decimal BeforeQuantity { get; set; } = 0;

    [SugarColumn(ColumnName = "AfterQuantity", ColumnDescription = "变动后数量", DefaultValue = "0")]
    public virtual decimal AfterQuantity { get; set; } = 0;

    [SugarColumn(ColumnName = "CostPrice", ColumnDescription = "成本单价", DefaultValue = "0")]
    public virtual decimal CostPrice { get; set; } = 0;

    [SugarColumn(ColumnName = "ChangeAmount", ColumnDescription = "变动金额", DefaultValue = "0")]
    public virtual decimal ChangeAmount { get; set; } = 0;

    [SugarColumn(ColumnName = "LocationCode", ColumnDescription = "库位编码", Length = 50)]
    public virtual string? LocationCode { get; set; }

    [SugarColumn(ColumnName = "BatchNo", ColumnDescription = "批号", Length = 50)]
    public virtual string? BatchNo { get; set; }

    [SugarColumn(ColumnName = "Remark", ColumnDescription = "备注", Length = 500)]
    public virtual string? Remark { get; set; }
}
